### DISCUSSION TITLE
How to set or obtain the detail info for observer options?

### DISCUSSION THREAD

**Discussion by tinaZC** [numbbo/coco#2274] [Q&A]
May I ask how to set and obtain the detail info for observer options, such as number_targed_triggers, Number_evaluation triggers, Target_precision. It seems can be set or obtained, but I couldn't find any relevant introduction in the help document.

**Comment by nikohansen** [comment]
The triggers are determined in the [C code](http://numbbo.github.io/coco-doc/C) where I can't give any hints out of my head. Data generated with a changed setup may not be compatible with archived data in particular when fewer targets would be recorded (for which it is hard to see a good reason).

The displayed targets can be manipulated via the [`cocopp`](https://numbbo.github.io/gforge/apidocs-cocopp/cocopp.html) module which has the `cocopp.genericsettings` submodule to be checked for its attribute settings and [`cocopp.testbedsettings`](https://numbbo.github.io/gforge/apidocs-cocopp/cocopp.testbedsettings.html) where displayed targets for the testbeds are defined. Specifically, the target triggers depend on the testbed. The evaluation triggers are generally not used for the performance assessment.

Out of my head I don't have the specifics, however [this issue](https://github.com/numbbo/coco/issues/1421#issuecomment-311636538) has a use case for changing the targets.

In some cases, `cocopp.config.config()` has to be called to activate changes, see e.g. #1832

@tinaZC it may be useful to tell us more specifically what your goal is?

**Comment by nikohansen** [comment]
There is a duplicate with the answer at #2275

**Comment by tinaZC** [comment]
Dear nikohansen，

Thank you for your reply, and ask my goal of the question.
According to the reply, I know that there is a difference between the logging and displaying target trigger. The logging trigger should not need to be changed, as it is already sufficient.
The reason why I want to change the number_targed_triggers is because we focus more on the performance of my optimization algorithm when the target accuracy is higher, such as from e-3 to e-8, and ignore the data when the target accuracy at number_targed_triggers = e+2. However, after modifying the settings in “testbedsetting”,  I see that there is not much change in the curve.
Could you tell me if my understanding is incorrect? 

Thanks again.

Tina.

**Comment by tinaZC** [comment]
Dear nikohansen，

Thank you for your reply, and ask my goal of the question.
According to the reply, I know that there is a difference between the logging and displaying target trigger. The logging trigger should not need to be changed, as it is already sufficient.
The reason why I want to change the number_targed_triggers is because we focus more on the performance of my optimization algorithm when the target accuracy is higher, such as from e-3 to e-8, and ignore the data when the target accuracy at number_targed_triggers = e+2. However, after modifying the settings in “testbedsetting”, I see that there is not much change in the curve.
Could you tell me if my understanding is incorrect?

Thanks again.

Tina.



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "numbbo/coco"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年4月24日(星期三) 晚上8:50
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [numbbo/coco] How to set or obtain the detail info for observer options? (Discussion #2274)





 
The triggers are determined in the C code where I can't give any hints out of my head. Data generated with a changed setup may not be compatible with archived data.
 
The displayed targets can be manipulated via the cocopp module which has the cocopp.genericsettings submodule to be checked for its attribute settings and cocopp.testbedsettings where displayed targets for the testbeds are defined.
 
Out of my head I don't have the specifics, however this issue has a use case.

**Comment by nikohansen** [comment]
Looks all right. For a "sanity check" you could produce single-target post-processing versions, like setting the list to contain only the last target `1e-8`. In this case you should see in particular less smooth and steeper empirical runtime distributions (and they should always be more to the right than the previous ones). Or with a single easy target like `1e-1`, the graphs should be more to the left and tend to go further up.

**Reply by tinaZC** [reply]
Thank you,  nikohansen.
